crud/get.lua

```diff
@@ -65,26 +62,25 @@ function get.call(space_name, key, opts)
     end
 
     local bucket_id = vshard.router.bucket_id_strcrc32(key)
-
-    local replicaset, err = vshard.router.route(bucket_id)
-    if replicaset == nil then
-        return nil, GetError:new("Failed to get replicaset for bucket_id %s: %s", bucket_id, err.err)
-    end
-
-    local results, err = call.rw(GET_FUNC_NAME, {space_name, key}, {
-        replicasets = {replicaset},
-        timeout = opts.timeout,
-    })
+    local results, err = call.rw_single(
+        bucket_id, GET_FUNC_NAME,
+        {space_name, key}, {timeout=opts.timeout})
 
     if err ~= nil then
         return nil, GetError:new("Failed to get: %s", err)
     end
 
-    local tuple = results[replicaset.uuid]
-    return {
-        metadata = table.copy(space:format()),
-        rows = {tuple},
-    }
+    if results == nil then
+        return {
+            metadata = table.copy(space:format()),
+            rows = {},
+        }
+    else
+        return {
+           metadata = table.copy(space:format()),
+           rows = {results},
+        }
+    end
```

Comment:
What about

```lua
if results == nil then 
    results = nil
end

return {
           metadata = table.copy(space:format()),
           rows = {results},
}
```

?

Reply:
This is because the `result` can be `box.NULL`, in this case, it will make it into the rows table. I will make a comment about that.

Reply:
Your `if` branch returns, why do you use `else`?

This code handles `box.NULL` case and makes code much more clear:

```
if results == nil then 
    results = nil
end
```